javascript/javascript1/week2/smartease.js

```diff
@@ -0,0 +1,177 @@
+  } else if (temperature > 5 && temperature < 18) {
+    return "Normal Jacket and Pant and T-shirt";
+  } else if (temperature >= 18) {
+    return "Shorts and a T-shirt";
+  } else {
+    return "Please provide the current temperature in degrees";
+  }
+}
+const temperature = 18;
+const clothesToWear = selectClothes(temperature);
+console.log(clothesToWear);
+
+//Student manager
+console.log("**********************");
+console.log("Ex 5: Student manager");
+console.log("**********************");
+let class07Students = [];
+function addStudentToClass(studentName) {
+  studentName = studentName.toLowerCase();
+  if (
+    (class07Students.length < 6 &&
+      !class07Students.includes(studentName) &&
+      studentName !== "") ||
+    studentName === "mary elizabeth"
+  ) {
+    class07Students.push(studentName);
+  } else if (studentName === "") {
+    console.log("Please provide the student name");
+  } else if (class07Students.includes(studentName)) {
+    console.log(`Student ${studentName} is already in the class`);
+  } else {
+    console.log("Cannot add more students to class 07");
+  }
+  return class07Students;
+}
+addStudentToClass("Benjamin");
+addStudentToClass("Hughes");
+addStudentToClass("Josefine");
+addStudentToClass("Henrik");
+addStudentToClass("Ida");
+addStudentToClass("Frede");
+addStudentToClass("Mary Elizabeth");
+addStudentToClass("Hello");
+addStudentToClass("benjamin");
+addStudentToClass("");
+
+console.log(class07Students);
+
+function getNumberOfStudents() {
+  return class07Students.length;
+}
+console.log(getNumberOfStudents());
+
+//Candy helper optional
+console.log("**********************");
+console.log("Ex 6: Candy helper optional");
+console.log("**********************");
+let price = 0;
+function addCandy(candyType, weight) {
+  candyType = candyType.toLowerCase();
+  if (candyType === "sweet") {
+    price = weight * 0.5;
+  } else if (candyType === "chocolate") {
+    price = weight * 0.7;
+  } else if (candyType === "toffee") {
+    price = weight * 1.1;
+  } else if (candyType === "chewing-gum") {
+    price = weight * 0.03;
+  } else {
+    console.log(
+      "Please select the candyType as Sweet or Chocolate or Toffee or Chewing-gum"
+    );
+  }
+  return (price = Math.round(price));
+}
+let boughtCandyPrices = [];
+addCandy("sweet", 15);
+boughtCandyPrices.push(price);
+addCandy("toffee", 20);
+boughtCandyPrices.push(price);
+addCandy("chocolate", 15);
+boughtCandyPrices.push(price);
+console.log(boughtCandyPrices);
+
+let amountSpent = 0;
+for (let i = 0; i < boughtCandyPrices.length; i++) {
+  amountSpent += boughtCandyPrices[i];
+}
+// using while loop
+/*let i = 0;
+while (i < boughtCandyPrices.length) {
+  amountSpent += boughtCandyPrices[i];
+  i++;
+}*/
+
+const amountToSpend = Math.round(Math.random() * 100);
+function canBuyMoreCandy() {
+  if (amountToSpend > amountSpent) {
+    console.log("Yes, You can buy more candy");
+  } else {
+    console.log("Enough candy for you!");
+  }
+}
```

Comment:
you function `canBuyMoreCandy` has no input parameters and also no return value. It depends on the outer scope completely. Can you make is such that it take the `boughtCandyPrices` as parameter, and calculates the sum of the prices within the function. Then you can also check whether there is more candy to be bought within the function.

Reply:
I have revised the code based on your comments. Thanks for your time.